amethyst_network/src/resources/net_event.rs

```diff
@@ -0,0 +1,79 @@
+//! The network events that are passed from machine to machine, and within the ECS event handling system.
+//! NetEvent are passed through the network
+//! NetOwnedEvent are passed through the ECS, and contains the event's source (remote connection, usually)
+
+use shrev::Event;
+use resources::*;
+
+///The basic network events shipped with amethyst
+///You can add more event by making a new enum and having this one has a variant of yours.
```

Comment:
Do you think it might be easier to go the other way and have our `NetEvent` enum have a variant called `CustomEvent(T)` where T is the provided enum?  That way we don't have to mess around with traits as much.  However it also means our NetEvent enum won't have a concrete type.  With the right generic coding this might not be a problem, but I leave that decision to you.